### PR TITLE
[ESPN] Include additional URL formats (fixes issue #13244)

### PR DESCRIPTION
**Pull request by parmjitv**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x ] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [ x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [ x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

Updated the regex for the [ESPN] extractor to include additional URL formats for videos featured on the site - issue #13244.

Cheers,

Parmjit V.

### CODE REVIEW

File: youtube_dl/extractor/espn.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 
 class ESPNIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:espn\.go|(?:www\.)?espn)\.com/video/clip(?:\?.*?\bid=|/_/id/)(?P<id>\d+)'
+    _VALID_URL = r'https?://(?:(?:(\w+\.)+)?espn\.go|(?:www\.)?espn)\.com/(?:video/clip(?:\?.*?\bid=|/_/id/)|watch/player\?.*?\bid=)(?P<id>\d+)'
```
Comment (dstftw) [inline youtube_dl/extractor/espn.py]: 1. `(\w+\.)` is captured for no purpose.
2. It's probably time to make regex verbose `(?x)`.

File: youtube_dl/extractor/espn.py [hunk 2]
```diff
@@ -25,15 +25,70 @@ class ESPNIE(InfoExtractor):
...
+            'description': 'md5:2b0f5bae9616d26fba8808350f0d2b9b',
+            'timestamp': 1489539155,
+            'upload_date': '20170315',
+        },
+        'params': {
+            'skip_download': True,
+        },
+        'expected_warnings': ['Unable to download f4m manifest'],
+    }, {
+        'url': 'http://nonredline.sports.espn.go.com/video/clip?id=19744672',
```
Comment (dstftw) [inline youtube_dl/extractor/espn.py]: `only_matching`.

File: youtube_dl/extractor/espn.py [hunk 3]
```diff
@@ -25,15 +25,70 @@ class ESPNIE(InfoExtractor):
...
+            'description': 'md5:2fe30038633f86408a184ec676c5fdac',
+            'timestamp': 1498519201,
+            'upload_date': '20170626',
+        },
+        'params': {
+            'skip_download': True,
+        },
+        'expected_warnings': ['Unable to download f4m manifest'],
+    }, {
+        'url': 'https://cdn.espn.go.com/video/clip/_/id/19771774',
```
Comment (dstftw) [inline youtube_dl/extractor/espn.py]: `only_matching`.

File: youtube_dl/extractor/espn.py [hunk 4]
```diff
@@ -25,15 +25,70 @@ class ESPNIE(InfoExtractor):
...
+            'description': 'md5:a60393eaee9c203dd4184c48c308651b',
+            'timestamp': 1498806684,
+            'upload_date': '20170630',
+        },
+        'params': {
+            'skip_download': True,
+        },
+        'expected_warnings': ['Unable to download f4m manifest'],
+    }, {
+        'url': 'http://www.espn.com/watch/player?id=19141491',
```
Comment (dstftw) [inline youtube_dl/extractor/espn.py]: `only_matching`.

File: youtube_dl/extractor/espn.py [hunk 5]
```diff
@@ -25,15 +25,70 @@ class ESPNIE(InfoExtractor):
...
+            'description': 'md5:9b31cf7b31e85c6dfcdefd11571e6c8e',
+            'timestamp': 1492011670,
+            'upload_date': '20170412',
+        },
+        'params': {
+            'skip_download': True,
+        },
+        'expected_warnings': ['Unable to download f4m manifest'],
+    }, {
+        'url': 'http://www.espn.com/watch/player?bucketId=257&id=19505875',
```
Comment (dstftw) [inline youtube_dl/extractor/espn.py]: `only_matching`.

**Comment by parmjitv** [comment]
I have applied the requested fixes, please review at your convenience. Thanks!

**Comment by parmjitv** [comment]
Shall I make any further corrections for this PR?  Thank you!

File: youtube_dl/extractor/espn.py [hunk 6]
```diff
@@ -10,7 +10,19 @@
...
-    _VALID_URL = r'https?://(?:espn\.go|(?:www\.)?espn)\.com/video/clip(?:\?.*?\bid=|/_/id/)(?P<id>\d+)'
+    _VALID_URL = r'''(?x)
+                    https?://
+                        (?:
+                            (?:(?:\w+\.)+)?espn\.go|
+                            (?:www\.)?espn
+                        )\.com/
+                        (?:
+                            video/clip(?:\?.*?\bid=|/_/id/)|
+                            watch/player\?.*?\bid=
```
Comment (dstftw) [inline youtube_dl/extractor/espn.py]: Does not cover http://www.espn.com/watch/player/_/id/19141491.

File: youtube_dl/extractor/espn.py [hunk 7]
```diff
@@ -10,7 +10,19 @@
...
-    _VALID_URL = r'https?://(?:espn\.go|(?:www\.)?espn)\.com/video/clip(?:\?.*?\bid=|/_/id/)(?P<id>\d+)'
+    _VALID_URL = r'''(?x)
+                    https?://
+                        (?:
+                            (?:(?:\w+\.)+)?espn\.go|
+                            (?:www\.)?espn
+                        )\.com/
+                        (?:
+                            video/clip(?:\?.*?\bid=|/_/id/)|
+                            watch/player(?:\?.*?\bid=|/_/id/)
```
Comment (dstftw) [inline youtube_dl/extractor/espn.py]: Code duplication.

**Comment by parmjitv** [comment]
Good catch - I have updated the regex and added this URL to the test suite for this extractor.  Please let me know if there are any other issues.  Thanks!

**Pull request merged**